pkg/raft/types.go

```diff
@@ -148,3 +148,53 @@ func (s logSlice) valid() error {
 	}
 	return nil
 }
+
+// snapshot is a state machine snapshot tied to the term of the leader who
+// observed this committed state.
+//
+// Semantically, from the log perspective, this type is equivalent to a logSlice
+// from 0 to lastEntryID(), plus a commit logMark. All leader logs / snapshots
+// at terms >= snapshot.term are consistent with this snapshot, i.e. all entries
+// up to the lastEntryID() match.
+type snapshot struct {
+	// term is the term of the leader log with whom the snapshot is consistent.
+	//
+	// Invariant: term >= term[generated] >= term[committed], where:
```

Comment:
This might sound a bit abstract, but here is an example where the `Term` in the message is arbitrarily bumped: https://github.com/cockroachdb/cockroach/blob/b71677c876b127f2dc775d34c561d71cd7814ead/pkg/kv/kvserver/replica_raft.go#L619-L631